### PR TITLE
Add CallToAction

### PR DESCRIPTION
**Pull request by MadKingBryce**
Added inputs/outputs

Still working on styling and layout.

### CODE REVIEW

**Comment by bherr2** [comment]
Please update the PR title

**Review by bherr2 — DISMISSED** [review]
Very close! Some simple changes needed and code smells to fix.

File: projects/ccf-shared/src/lib/components/call-to-action/call-to-action.component.scss [hunk 1]
```diff
@@ -0,0 +1,71 @@
+.content{
```
Comment (axdanbol) [inline projects/ccf-shared/src/lib/components/call-to-action/call-to-action.component.scss:1]: Move all styles into a top level `:host` selector. Move `.content` styles into the top level `:host`.

File: projects/ccf-shared/src/lib/components/call-to-action/call-to-action.component.scss [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  display: block;
+  align-content: center;
+  padding-left: 1.6rem;
+  padding-right: 1.6rem;
+}
+
+.info-button-wrapper {
```
Comment (axdanbol) [inline projects/ccf-shared/src/lib/components/call-to-action/call-to-action.component.scss:8]: I'm unable to locate this class in the html -> please remove

File: projects/ccf-shared/src/lib/components/call-to-action/call-to-action.component.html [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<div class="rui-about wrapper">
+  <div class="container">
+    <div class="content">
+      <div class="firstRow">
+        <p class="title" mat-dialog-title>{{infoTitle}}</p>
+        <mat-icon class="material-icons close-icon" (click)="close()">clear</mat-icon>
+      </div>
+      <img mat-card-sm-image [src]="imageUrl" alt="preview" layout-fill>
```
Comment (axdanbol) [inline projects/ccf-shared/src/lib/components/call-to-action/call-to-action.component.html:8]: Add a class to this `img` element

File: projects/ccf-shared/src/lib/components/call-to-action/call-to-action.component.scss [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  font-size: 14px;
+  padding-right: 1rem;
+  padding-top: .75rem;
+}
+
+.firstRow {
+  display: flex;
+  width: 100%;
+  line-height: .9rem;
+  p {
```
Comment (axdanbol) [inline projects/ccf-shared/src/lib/components/call-to-action/call-to-action.component.scss:31]: Use a class selector instead

File: projects/ccf-shared/src/lib/components/call-to-action/call-to-action.component.scss [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    padding-right: 7.0rem;
+    padding-left: .2rem;
+    
+  }
+  mat-icon {
```
Comment (axdanbol) [inline projects/ccf-shared/src/lib/components/call-to-action/call-to-action.component.scss:36]: Use a class selector instead

File: projects/ccf-shared/src/lib/components/call-to-action/call-to-action.component.scss [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    line-height: 3.4rem;
+  }
+
+}
+
+.message{
+  padding: .5rem;
+}
+
+img {
```
Comment (axdanbol) [inline projects/ccf-shared/src/lib/components/call-to-action/call-to-action.component.scss:47]: Use a class selector instead

File: projects/ccf-shared/src/lib/services/local-storage/local-storage.service.ts [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        error.name === 'QuotaExceededError' ||
+        // Firefox
+        error.name === 'NS_ERROR_DOM_QUOTA_REACHED'
+      ) && (storage && storage.length !== 0);
+
+      return full ? storage : undefined;
+    }
+  })();
+
+  get length(): number {
```
Comment (axdanbol) [inline projects/ccf-shared/src/lib/services/local-storage/local-storage.service.ts:35]: Missing code docs

File: projects/ccf-shared/src/lib/states/call-to-action/call-to-action.state.ts [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+import { GoogleAnalyticsService } from 'ngx-google-analytics';
+import { Observable } from 'rxjs';
+import { map, tap } from 'rxjs/operators';
+
+import { CallToActionBehaviorComponent } from '../../components/call-to-action-behavior/call-to-action-behavior.component';
+import { DocumentationContent, InfoButtonService, PanelData } from '../../components/info/info-button/info-button.service';
+import { InfoDialogComponent } from '../../components/info/info-dialog/info-dialog.component';
+import { LocalStorageService } from '../../services/local-storage/local-storage.service';
+import { CloseDialog, LearnMore, OpenDialog } from './call-to-action.actions';
+
```
Comment (axdanbol) [inline projects/ccf-shared/src/lib/states/call-to-action/call-to-action.state.ts:14]: This file needs code documentation

**Review by axdanbol — CHANGES_REQUESTED** [review]
A few html and scss changes

File: projects/ccf-shared/src/lib/components/call-to-action/call-to-action.component.html [hunk 9]
```diff
@@ -0,0 +1,16 @@
+<div class="rui-about wrapper">
```
[the remark this answers is not in the document]
Comment (MadKingBryce) [reply projects/ccf-shared/src/lib/components/call-to-action/call-to-action.component.html:1]: Removing any of these significantly alters the resulting dialog box.

File: projects/ccf-shared/src/lib/components/call-to-action/call-to-action.component.scss (hunk 1 above)
Comment (MadKingBryce) [reply projects/ccf-shared/src/lib/components/call-to-action/call-to-action.component.scss:1]: Removing the content class significantly changes the borders of the dialog box.

File: projects/ccf-shared/src/lib/components/call-to-action/call-to-action.component.scss [hunk 10]
```diff
@@ -0,0 +1,62 @@
+:host{
```
Comment (axdanbol) [inline projects/ccf-shared/src/lib/components/call-to-action/call-to-action.component.scss:1]: Please convert px values to rem

**Review by axdanbol — CHANGES_REQUESTED** [review]
Convert px to rem

**Review by axdanbol — APPROVED** [review]

**Pull request merged**